File: NeetCode150/ContainsDups.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # Solution 1 
        seen = set()
        for x in nums:
            if x in seen:
                return True
            seen.add(x)
        return False

        # Solution 2: One liner
        # return len(set(nums)) != len(nums)

File: NeetCode150/Array&Hashing/ContainsDups.cpp
#include <set>
#include <algorithm>

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {

        sort(nums.begin(), nums.end());
        for (int x = 1; x < nums.size(); x++) {
            if (nums[x] == nums[x - 1]) {
                return true;
            }
        }

        // set<int> seen;
        // for (auto &x: nums) {
        //     if (seen.find(x) != seen.end()) {
        //         return true;
        //     }
        //     seen.insert(x);
        // }        

        return false;
    }
};
